src/Core/Domain/AttributeGroup/ValueObject/AttributeGroupId.php

```diff
@@ -40,45 +42,34 @@ final class AttributeGroupId
 
     /**
      * @param int $attributeGroupId
-     *
-     * @throws AttributeGroupConstraintException
      */
-    public function __construct($attributeGroupId)
+    public function __construct(int $attributeGroupId)
     {
-        $this->assertIsIntegerGreaterThanZero($attributeGroupId);
-        $this->attributeGroupId = $attributeGroupId;
-    }
+        $this->assertIntegerIsGreaterThanZero($attributeGroupId);
 
-    /**
-     * @deprecated
-     * @see getValue()
-     *
-     * @return int
-     */
-    public function getAttributeGroupId()
-    {
-        return $this->attributeGroupId;
+        $this->attributeGroupId = $attributeGroupId;
     }
 
     /**
      * @return int
      */
-    public function getValue()
+    public function getValue(): int
     {
         return $this->attributeGroupId;
     }
 
     /**
-     * Validates that the value is integer and is greater than zero
-     *
-     * @param int $value
+     * @param int $attributeGroupId
      *
      * @throws AttributeGroupConstraintException
      */
-    private function assertIsIntegerGreaterThanZero($value)
+    private function assertIntegerIsGreaterThanZero(int $attributeGroupId): void
     {
-        if (!is_int($value) || 0 >= $value) {
-            throw new AttributeGroupConstraintException(sprintf('Invalid attribute group id "%s".', var_export($value, true)), AttributeGroupConstraintException::INVALID_ID);
+        if (0 > $attributeGroupId) {
```

Comment:
```suggestion
        if (0 >= $attributeGroupId) {
```
Zero should not be a valid value